lib/api/core/validation/types/geoShape.js

```diff
@@ -0,0 +1,282 @@
+ * @returns {boolean}
+ */
+GeoShapeType.prototype.checkStructure = function (allowedShapes, shape, errorMessages) {
+  var result = true;
+  if (!this.checkAllowedProperties(shape, allowedShapeProperties)) {
+    errorMessages.push('The shape object has a not allowed property.');
+    result = false;
+  }
+
+  if (!shape.type) {
+    errorMessages.push('The shape object has no type defined.');
+    result = false;
+  }
+
+  if (allowedShapes.indexOf(shape.type) === -1) {
+    errorMessages.push('The provided shape type is not allowed.');
+    result = false;
+  }
+
+  if (shape.coordinates && shape.type === 'geometrycollection' && !Array.isArray(shape.coordinates)) {
+    errorMessages.push('The coordinates property must not be provided for the "geometrycollection" shape type.');
+    result = false;
+  }
+
+  if (!shape.coordinates && shape.type !== 'geometrycollection') {
+    errorMessages.push(`The coordinates property must be provided for the "${shape.type}" shape type.`);
+    result = false;
+  }
+
+  if (shape.type === 'circle' && !shape.radius) {
+    errorMessages.push('The radius property is mandatory for the "circle" shape type.');
+    result = false;
+  }
+
+  if (shape.radius && shape.type !== 'circle') {
+    errorMessages.push(`The radius property must not be provided for the "${shape.type}" shape type.`);
+    result = false;
+  }
+
+  if (shape.orientation && shape.type !== 'polygon' && shape.type !== 'multipolygon') {
+    errorMessages.push(`The orientation property must not be provided for the "${shape.type}" shape type.`);
+    result = false;
+  }
+
+  if (shape.geometries && shape.type !== 'geometrycollection') {
+    errorMessages.push(`The geometries property must not be provided for the "${shape.type}" shape type.`);
+    result = false;
+  }
+
+  if (shape.type === 'geometrycollection' && !shape.geometries) {
+    errorMessages.push('The geometries property must be provided for the "geometrycollection" shape type.');
+    result = false;
+  }
+
+  return result;
+};
+
+/**
+ * @param {TypeOptions} typeOptions
+ * @return {boolean|TypeOptions}
+ * @throws InternalError
+ */
+GeoShapeType.prototype.validateFieldSpecification = function (typeOptions) {
+  var result = true;
+
+  if (typeOptions.hasOwnProperty('shapeTypes')) {
+    if (!Array.isArray(typeOptions.shapeTypes) || typeOptions.shapeTypes.length === 0) {
+      return false;
+    }
+
+    typeOptions.shapeTypes.forEach(shape => {
+      if (allowedShapeTypes.indexOf(shape) === -1) {
+        result = false;
+      }
+    });
+
+    if (!result) {
+      return false;
+    }
+  }
+  else {
+    typeOptions.shapeTypes = allowedShapeTypes;
+  }
+
+  return typeOptions;
+};
+
+/**
+ * @param {GeoShapePointCoordinates} point
+ * @returns {boolean}
+ */
+function isPoint (point) {
+  if (!Array.isArray(point) || point.length !== 2) {
+    return false;
+  }
+
+  return !(point[0] < -180 || point[0] > 180 || point[1] < -90 || point[1] > 90);
```

Comment:
What Benoit meant is that you're testing for a point with its first coordinate between ]-180,180[ and its second one between ]-90, 90[
This looks like a "lon, lat" point, not a "lat, lon" one.